File: Amma/Services/FamilyContext.swift
import Foundation

final class FamilyContext {
    static let shared = FamilyContext()

    let familyId: UUID

    private init() {
        let key = "familyId"
        if let stored = UserDefaults.standard.string(forKey: key), let uuid = UUID(uuidString: stored) {
            familyId = uuid
        } else {
            let generated = UUID()
            UserDefaults.standard.set(generated.uuidString, forKey: key)
            familyId = generated
        }
    }
}

File: android/app/src/main/java/com/dhana/amma/services/FamilyContext.kt
package com.dhana.amma.services

import android.content.Context
import java.util.UUID

class FamilyContext(context: Context) {
    private val prefs = context.getSharedPreferences("amma_family", Context.MODE_PRIVATE)

    val familyId: UUID by lazy {
        val stored = prefs.getString(KEY_FAMILY_ID, null)
        if (stored != null) {
            UUID.fromString(stored)
        } else {
            val generated = UUID.randomUUID()
            prefs.edit().putString(KEY_FAMILY_ID, generated.toString()).apply()
            generated
        }
    }

    private companion object {
        const val KEY_FAMILY_ID = "family_id"
    }
}
